webapi/src/main/scala/org/knora/webapi/messages/util/search/gravsearch/transformers/OntologyInferencer.scala

```diff
@@ -33,88 +35,57 @@ final case class OntologyInferencer(private val ontologyCache: OntologyCache)(im
         case other =>
           ZIO.fail(GravsearchException(s"The object of rdf:type must be an IRI, but $other was used"))
       }
-
+  } yield {
     // look up subclasses from ontology cache
-    superClasses = cache.classToSubclassLookup
-    knownSubClasses =
-      superClasses
-        .get(baseClassIri.iri)
-        .getOrElse({
-          Set(baseClassIri.iri)
-        })
-        .toSeq
+    val knownSubClasses = cache.classToSubclassLookup.get(baseClassIri.iri).getOrElse(Set(baseClassIri.iri)).toSeq
 
     // if provided, limit the child classes to those that belong to relevant ontologies
-    subClasses = limitInferenceToOntologies match {
-                   case None                       => knownSubClasses
-                   case Some(relevantOntologyIris) =>
-                     // filter the known subclasses against the relevant ontologies
-                     knownSubClasses.filter { subClass =>
-                       cache.classDefinedInOntology.get(subClass) match {
-                         case Some(ontologyOfSubclass) =>
-                           // return true, if the ontology of the subclass is contained in the set of relevant ontologies; false otherwise
-                           relevantOntologyIris.contains(ontologyOfSubclass)
-                         case None => false // should never happen
-                       }
-                     }
-                 }
+    val subClasses = limitInferenceToOntologies match {
+      case None                       => knownSubClasses
+      case Some(relevantOntologyIris) =>
+        // filter the known subclasses against the relevant ontologies
+        knownSubClasses.filter(cache.classDefinedInOntology.get(_).exists(relevantOntologyIris.contains(_)))
+    }
+
     // if subclasses are available, create a union statement that searches for either the provided triple (`?v a <classIRI>`)
     // or triples where the object is a subclass of the provided object (`?v a <subClassIRI>`)
     // i.e. `{?v a <classIRI>} UNION {?v a <subClassIRI>}`
```

Comment:
comment should either be removed or updated

Reply:
Take a look.